### DISCUSSION TITLE
Simple question regarding Artillery

### DISCUSSION THREAD

**Discussion by Evengelius** [artilleryio/artillery#1316] [Artillery]
Hello everyone,

I've got a simple question regarding Artillery,

I was wondering, is Artillery the right tool if what I want to achieve is load-testing **a web-application** on a **production or UAT** environment?
I know that this tool works great with mobile applications, but I don't know if it is the same nor the right tool for web-apps.

For example as a right tool, I know that there is Gatling, and it has a Recorder.
But what about Artillery?

Kind regards.

**Comment by hassy** [comment]
hi @Evengelius 👋 yes, Artillery is very well suited for testing web apps.

You can write test scripts that talk to the backend routes/endpoints, or you can run full browser tests with: https://github.com/artilleryio/artillery-engine-playwright

If you use the Playwright integration, you can use Playwright's built-in codegen tool to record web page activity and replay it with Artillery: https://playwright.dev/docs/cli#generate-code

**Comment by Evengelius** [comment]
Hey @hassy ,

Thank you very much for your quick answer !

So, if my page doesn't return a JSON but an HTML with all the CSS & XPATH selectors there, it's better to use PlayWright.

Thus I was wondering, concerning this solution, is it possible to set a number of users we want to run the test for ? Like feeders in Gatling ?

Like for instance we have a Login scenario with "username" and "password" field,

Is it possible to have like a CSV file that contains these fields as parameters, and that it loops through these parameters for each user,

`username, password`
`john, doe`
`jane, doe`

Running the Login the scenario for each one of them? 

-----

Edit : 

I saw that we pass our PlayWright script in the YML processor field as shown here : 

```
  engines:
    playwright: {}
  processor: ./login.js
```

And that the payload and its fields parameters goes here :

```
  payload:
    path: "./loginForm.csv"
    fields:
      - "username"
      - "password
```

So does it means that it is possible to link these "username" & "password" fields from the payload, with the "username" & "password" fields from the processor ?

So that it will run the PlayWright script "login.js", and log-in with the credentials that are inside of the CSV file?

**Reply by hassy** [reply]
hi @Evengelius! yep, `payload` option is exactly it. Values from `username` and `password` fields are available inside processor functions from `vuContext.vars` object, e.g. with `vuContext.vars.username`, see the second example here: https://github.com/artilleryio/artillery-engine-playwright#flow-function-api

**Reply by Evengelius** [reply]
Thank you very much @hassy it works , you really built something incredible, and mixing it with Playwright was brilliant idea.
